src/classes/BusinessLogicServices/CarrierService.php

```diff
@@ -669,4 +716,26 @@ private function getPrestaCarrierLogoPath($id)
 
         return rtrim($imgDir, '/') . '/' . $id . '.jpg';
     }
+
+    /**
+     * Gets non-Packlink carriers.
+     *
+     * @param string $select
+     *
+     * @return array
+     *
+     * @throws \PrestaShopDatabaseException
+     */
+    private function getNonPacklinkCarriers($select = 'id_carrier')
```

Comment:
I think this query should be place in other layer . I think the service should be agnostic the store

Reply:
This service is dedicated to PrestaShop carrier services. Usually, a query would go to the repository, but in PrestaShop, there is only this generic Db singleton for working with the database. This is why it is placed here.

Reply:
Ok